.github/workflows/first-time-setup.yaml

```diff
@@ -0,0 +1,116 @@
+name: first-time-setup
+run-name: first time setup of repo
+
+on:
+  # run when called from another workflow
+  workflow_call:
+
+  # run if user manually requests it
+  workflow_dispatch:
+
+permissions:
+  contents: write
+
+jobs:
+  first-time-setup:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Print contexts
+        uses: crazy-max/ghaction-dump-context@v1
+
+      - name: Create Pages branch
+        uses: peterjgrainger/action-create-branch@v2.4.0
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          branch: "gh-pages"
+
+      - name: Checkout Pages branch
+        uses: actions/checkout@v3
+        with:
+          ref: gh-pages
+
+      - name: Clear Pages branch
+        run: rm -rf * .github .gitignore
+
+      - name: Make .nojekyll file
+        run: touch .nojekyll
+
+      - name: Make placeholder homepage
+        run: echo "Your site will appear here" > index.html
```

Comment:
> if the message mentioned that a build process was going on

Technically this isn't always true. The user has to manually run the first time set up, then they have to manually enable GitHub Pages, and only then does the site start building. So I don't want to mislead them into thinking the build is already happening. Side note, I used to have Pages be enabled automatically from the same workflow, but GitHub discovered a security vulnerability and removed that capability from its API.

I'm not sure what placeholder text, if any, would be appropriate here. I did want to commit _some_ file to the `gh-pages` branch just to distinguish that the first time setup was successful, and also to serve as a blank page while Pages is building (rather than a 404, which doesn't distinguish between "building" and "building not started yet").

Suggestions welcome.

Reply:
I don't know what the appropriate placeholder text should be, but IMHO it's better to see that message than nothing, so maybe it's fine as it is. It's common practice for web servers and content-management systems to show a "you set it up, but you still need to add content!" message, e.g. in Apache and Nginx, or Wordpress.

Reply:
Maybe just "Placeholder homepage"...